egs/chime4/ConvTasNet/local/create_metadata.py

```diff
@@ -0,0 +1,106 @@
+import os
+import argparse
+from glob import glob
+import pandas as pd
+import numpy as np
+
+# Command line arguments
+parser = argparse.ArgumentParser()
+parser.add_argument("--chime3_dir", type=str, default=None, help="Path to CHiME3 root directory")
+
+# Set seed for random generation
+SEED = 4
+np.random.seed(SEED)
+
+
+def main(args):
+    chime3_dir = args.chime3_dir
+    create_local_metadata(chime3_dir)
+
+
+def create_local_metadata(chime3_dir):
+    # Get CHiME-3 annotation files
+    c3_annot_files = [
+        f for f in glob(os.path.join(chime3_dir, "data", "annotations", "*real*.json"))
+    ]
+    # Get CHiME-4 annotation files
+    c4_annot_files = [
+        f for f in glob(os.path.join(chime3_dir, "data", "annotations", "*real*.list"))
+    ]
+    for c3_annot_file_path in c3_annot_files:
+        # Read CHiME-3 annotation file
+        c3_annot_file = pd.read_json(c3_annot_file_path)
+        # subsets : "tr" "dt" "et" origin "real" or "simu"
+        subset, origin = os.path.split(c3_annot_file_path)[1].replace(".json", "").split("_")
```

Comment:
I think the entire code could benefit from using the `pathlib` API. Eg here, `c3_annot_file_path.with_suffix("").name.split("_")`

Also I think it could be helpful to add a comment with a file name example everywhere filenames are parsed or constructed, for example

```
# Extract subset and origin from /foo/bar/<subset>_<origin>.json
```